grove/simon/simon.py

```diff
@@ -0,0 +1,424 @@
+"""Module for the Simon's Algorithm.
+For more information, see
+
+- http://lapastillaroja.net/wp-content/uploads/2016/09/\
+Intro_to_QC_Vol_1_Loceff.pdf
+- http://pages.cs.wisc.edu/~dieter/Courses/2010f-CS880/Scribes/05/lecture05.pdf
+"""
+
+import numpy as np
+import pyquil.quil as pq
+from pyquil.gates import *
+
+
+def unitary_function(mappings):
```

Comment:
I agree, as this is something that both this and deutsch_jozsa use. I have a general implementation of these oracle/unitary methods on my own fork (https://github.com/joelin0/grove/blob/blackbox/grove/blackbox_algorithms/utils.py), although it is not at all refined. But I agree, it would be nice to have separately in some blackbox utils file.